### PR TITLE
allows disable pod events enrichment with deployment name

### PR DESCRIPTION
**Pull request by newly12**
## What does this PR do?

allows disable pod events enrichment with deployment name

## Why is it important?

allows user to disable this enrichment due to the potential rate limit #28149 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have made corresponding change to the default configuration files
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.

## Author's Checklist

- [ ]

## Related issues

- Relates #28149

### CODE REVIEW

**Review by ChrsMark — COMMENTED** [review]
Hey @newly12 ! This is looking good! I just left 2 comments to consider.

File: metricbeat/module/kubernetes/util/kubernetes.go [hunk 1]
```diff
@@ -139,7 +139,7 @@ func NewResourceMetadataEnricher(
 	cfg, _ := common.NewConfigFrom(&metaConfig)
 
 	metaGen := metadata.NewResourceMetadataGenerator(cfg, watcher.Client())
-	podMetaGen := metadata.NewPodMetadataGenerator(cfg, nil, watcher.Client(), nil, nil)
+	podMetaGen := metadata.NewPodMetadataGenerator(cfg, nil, watcher.Client(), nil, nil, &metadata.AddResourceMetadataConfig{Deployment: true})
```
[the remark this answers is not in the document]
Comment (newly12) [reply metricbeat/module/kubernetes/util/kubernetes.go:142]: lol I was simply following same like the other arguments and meantime keep the same behavior like previous version. so current configuration doesn't have a section for namespace nor node, wondering if deployment should be added first there..

**Comment by newly12** [comment]
@ChrsMark updated per your suggestion.

File: metricbeat/module/kubernetes/util/kubernetes.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (newly12) [reply metricbeat/module/kubernetes/util/kubernetes.go]: in the later commit I still passed a `true` to this function for the comment. we can discuss more on how this one should be handled properly.

[the remark this answers is not in the document]
Comment (ChrsMark) [reply metricbeat/module/kubernetes/util/kubernetes.go]: Ah ok, I see! I think we can leave it as is for now and try to address it in general in another round. I will open an issue for this :).

**Review by ChrsMark — APPROVED** [review]
lgtm! Thanks for contributing this!

**Comment by ChrsMark** [comment]
@newly I see some lint errors, could you take a look?

**Comment by ChrsMark** [comment]
Maybe we could merge the latest upstream/master and check if it helps?

**Comment by ChrsMark** [comment]
CI passed, merging! Thank you @newly12 !

**Pull request merged**